src/python/tests/gff3/test_simplifier.py

```diff
@@ -0,0 +1,457 @@
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.gff3.restructure` module."""
+
+from contextlib import nullcontext as does_not_raise
+from os import PathLike
+from pathlib import Path
+from typing import Callable, ContextManager, Optional
+
+from Bio.SeqFeature import SeqFeature
+import pytest
+from pytest import param, raises
+
+from ensembl.io.genomio.gff3.exceptions import GFFParserError
+from ensembl.io.genomio.gff3.simplifier import GFFSimplifier
+
+
+def check_one_feature(input_gff: PathLike, output_gff: PathLike, check_function: str) -> SeqFeature:
+    """Load 1 feature from a GFF, apply a function, then write it back to a GFF."""
+    simp = GFFSimplifier()
+    simp.records.from_gff(input_gff)
+    # Get the only feature
+    feat = simp.records[0].features[0]
+    # Apply the named function
+    check_method = getattr(simp, check_function)
+    new_feat = check_method(feat)
+
+    if new_feat is not None:
+        # Put it back
+        if isinstance(new_feat, list):
+            simp.records[0].features = new_feat
+        else:
+            simp.records[0].features = [new_feat]
+        simp.records.to_gff(output_gff)
+    return new_feat
+
+
+@pytest.mark.parametrize(
+    "in_gff, expected_gff",
+    [
+        param("ok_gene.gff", "ok_gene.gff", id="ok gene"),
+        param("lone/transcript.gff", "lone/transcript_simped.gff", id="lone transcript"),
+        param("lone/trna.gff", "lone/trna_simped.gff", id="lone tRNA"),
+        param("lone/rrna.gff", "lone/rrna_simped.gff", id="lone rRNA"),
+    ],
+)
+def test_create_gene_for_lone_transcript(
+    tmp_path: Path,
+    data_dir: Path,
+    assert_files: Callable,
+    in_gff: PathLike,
+    expected_gff: PathLike,
+) -> None:
+    """Test gene create gene for lone transcript."""
+    input_gff = data_dir / in_gff
+    output_gff = tmp_path / Path(in_gff).name
+    new_feat = check_one_feature(input_gff, output_gff, "create_gene_for_lone_transcript")
+    if new_feat:
+        assert_files(output_gff, Path(data_dir / expected_gff))
```

Comment:
You're right, the underlying culprit `simpler_gff3_feature()` should always return a `SeqFeature`, and raise an exception otherwise. `Now check_one_feature()` do not return anything anymore so I've removed the `if new_feat` parts.